src/main-process/atom-application.js

```diff
@@ -227,11 +230,16 @@ class AtomApplication extends EventEmitter {
     this.applicationMenu = new ApplicationMenu(this.version, this.autoUpdateManager)
     this.atomProtocolHandler = new AtomProtocolHandler(this.resourcePath, this.safeMode)
 
-    this.listenForArgumentsFromNewProcess()
+    // Don't await for the following method to avoid delaying the opening of a new window.
+    // (we await it just after opening it).
```

Comment:
👍 Excellent.

Reply:
Again, worth mentioning explicitly that this method is slow on Windows?